src/libs/ajax/DataRepo.ts

```diff
@@ -142,11 +195,28 @@ export interface JobModel {
   class_name?: string;
 }
 
+export interface GetConceptsResponse {
+  result: SnapshotBuilderConcept[];
+}
+
+export interface GetConceptHierarchyResponse {
+  readonly result: SnapshotBuilderParentConcept[];
+}
+
+export interface SearchConceptsResponse {
+  result: SnapshotBuilderConcept[];
+}
+
+export interface SnapshotBuilderParentConcept {
+  parentId: number;
+  children: SnapshotBuilderConcept[];
+}
+
 export interface DataRepoContract {
   dataset: (datasetId: string) => {
     details: (include?: DatasetInclude[]) => Promise<DatasetModel>;
     roles: () => Promise<string[]>;
-    createSnapshotRequest(request: DatasetAccessRequest): Promise<DatasetAccessRequestApi>;
+    createSnapshotRequest(request: DatasetAccessRequestUI): Promise<DatasetAccessRequest>;
```

Comment:
This translation from the UI to the API type is confusing to me. Why can't we use the DatasetAccessRequest in both spots?

Reply:
I think one of the big reasons is that we have wanted additional information on the UI that doesn't make sense to send up to the API/have the API accept. However, it is weird that that is at the request level. I would expect the request object we build up to always match the API version, and do the conversions as we build up the request.

Reply:
Gotcha, make sense. I'm curious -  @evansuslovich  - What happens if you remove the DatasetAccessRequestUI alias here?

Reply:
Yes, if we remove `DatasetAccessRequestUI` and replace it with `DatasetAccessRequest` there will be issues because there are extra fields in the UI type, we can see that there is a `convertDatasetAccessRequest` which converts the `cohorts and `valueSets` with the `convertCohort` and `convertValueSet` functions.

```
export const convertDatasetAccessRequest = (datasetAccessRequest: DatasetAccessRequest) => {
  return {
    name: datasetAccessRequest.name,
    researchPurposeStatement: datasetAccessRequest.researchPurposeStatement,
    datasetRequest: {
      cohorts: _.map(convertCohort, datasetAccessRequest.datasetRequest.cohorts),
      conceptSets: datasetAccessRequest.datasetRequest.conceptSets,
      valueSets: _.map(convertValueSet, datasetAccessRequest.datasetRequest.valueSets),
    },
  };
};
```

Reply:
I think we could use the convert methods earlier, and thus do without the extra UI type, since we don't need that anywhere. That would allow us to do without the `DatasetAccessRequestUI` type, since we never construct it with the extra fields in the first place.

Reply:
Yes, I agree. I've removed all the types and I've converted at every instance of calling DataRepo functions